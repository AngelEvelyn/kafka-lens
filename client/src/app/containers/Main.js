const React = require('react');

//import components here
import ConnectionPage from './ConnectionPage';
import TopicPage from './TopicPage';
import '../css/index.css';

class Main extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      connected: false,
      uri_input: '',
      validString: 'Hello'
    };

    // bind methods here
    this.validConnectionChecker = this.validConnectionChecker.bind(this);
    this.updateURI = this.updateURI.bind(this);
  }
  // Lifecycle methods
  componentDidMount() {
    //code here
  }
  // Methods
  validConnectionChecker(event) {
    //code here
    event.preventDefault();
    if (this.state.validString === this.state.uri_input) {
      return this.setState({
        connected: true
      });
    }
  }

  updateURI(event) {
    const input = event.target.value;
    this.setState(state => {
      return { uri_input: input };
    });
  }

  render() {
    return (
      <div>
        <div>
          {this.state.connected === true ? (
            <TopicPage />
          ) : (
            <ConnectionPage
              validConnectionChecker={this.validConnectionChecker}
              updateURI={this.updateURI}
            />
          )}
        </div>
      </div>
    );
  }
}

export default Main;
