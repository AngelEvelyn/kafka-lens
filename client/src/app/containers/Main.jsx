import { ipcRenderer } from 'electron';

// import components here
import ConnectionPage from './ConnectionPage.jsx';
import TopicPage from './TopicPage.jsx';

import '../css/index.css';

const React = require('react');

class Main extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      connected: null,
      uri_input: '',
      topics: [],
      isFetching: false
    };

    // bind methods here
    this.validConnectionChecker = this.validConnectionChecker.bind(this);
    this.updateURI = this.updateURI.bind(this);
  }

  // Lifecycle methods
  componentWillMount() {
    // code here
    ipcRenderer.on('topic:getTopics', (e, data) => {
      this.setState({ isFetching: false });
      if (data === 'Error') {
        this.setState({
          connected: false
        });
      } else {
        this.setState({
          topics: data,
          connected: true
        });
      }
      console.log('logging state', this.state);
    });
  }

  // Methods
  validConnectionChecker(event) {
    event.preventDefault();
    this.setState({
      isFetching: true
    });

    if (this.state.uri_input === 'a') {
      const uri = '157.230.166.35:9092';
      ipcRenderer.send('topic:getTopics', uri);
    } else {
      ipcRenderer.send('topic:getTopics', this.state.uri_input);
    }
  }

  updateURI(event) {
    const input = event.target.value;
    this.setState(state => ({ uri_input: input }));
  }

  render() {
    const divStyle = {
      height: '100%'
    };
    return (
      <div style={divStyle}>
        <div style={divStyle}>
          {this.state.connected === true ? (
            <TopicPage topicList={this.state.topics} />
          ) : (
            <ConnectionPage
              validConnectionChecker={this.validConnectionChecker}
              updateURI={this.updateURI}
              connected={this.state.connected}
              isFetching={this.state.isFetching}
            />
          )}
        </div>
      </div>
    );
  }
}

export default Main;
