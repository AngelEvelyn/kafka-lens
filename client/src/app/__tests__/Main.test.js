import React from 'react';
import { shallow } from 'enzyme';

import Main from '../containers/Main.jsx';

describe('Main.js unit tests', () => {
  let wrapper;

  // * MOUNT MAIN COMPONENT BEFORE ALL TESTS
  beforeAll(() => {
    wrapper = shallow(<Main />);
  });

  it('should render without errors', () => {
    expect(wrapper.exists('.main-div')).toEqual(true);
    expect(wrapper).toMatchSnapshot();
    console.log(wrapper.debug());
  });

  it('should render a single div with class: main-div', () => {
    expect(wrapper.exists('.main-div')).toEqual(true);
  });

  it('should render <ConnectionPage/>', () => {
    expect(wrapper.exists('ConnectionPage')).toEqual(true);
  });

  it('should render <TopicPage/> if this.state.connected is true', () => {
    const instance = wrapper.instance();
    expect(instance.state.connected).toBe(null);
    instance.setState({ connected: true });
    expect(instance.state.connected).toBe(true);
    expect(wrapper.exists('TopicPage')).toEqual(true);
  });
});
