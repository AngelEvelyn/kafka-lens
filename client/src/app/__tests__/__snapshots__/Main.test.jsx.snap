// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main.js unit tests should render without errors 1`] = `
<div
  className="main-div"
>
  <ConnectionPage
    connected={null}
    isFetching={false}
    updateURI={[Function]}
    validConnectionChecker={[Function]}
  />
</div>
`;
